### ISSUE TITLE
Invalid extension context menu actions

### ISSUE THREAD

**Issue by Davoleo** [YellowFish085/al-search#32]
when right clicking the extension button these 5 options appear:
![image](https://github.com/user-attachments/assets/896af6bd-786a-418b-be97-cd098ec53dda)
although when clicked they just open the search page with "undefined" as query.

Also wanted to add that after v2 upgrade I had to log out and login again for AL-Search to work again, otherwise every time a query to AL was performed it would throw an error like these:
![image](https://github.com/user-attachments/assets/aa1aa34e-7ed1-4e07-90de-dfb1022d9768)
![image](https://github.com/user-attachments/assets/fe518106-38e5-4a79-9039-7ba6e4857f7d)
_(idk where I can find the actual extension logs)_
I managed to fix this by logging out and re-authorizing the app, but it might be a good idea to  warn the user about this via a CTA, otherwise they might think it's broken.

**Comment by YellowFish085** [comment]
Hello there, thanks for the feedback.


First thing, it looks like you're using Vivaldi, chromium based browsers are _kinda_ supported because you can install chrome extension on them but I don't "officially" support anything else than Chrome and Firefox so I'll try my best here.

I've just installed a fresh Vivaldi on my computer, kept all the settings as-is and just installed the add-on and I can't reproduce the contextual menu issue as all links properly display the selected text.

As for the login, I can't test it since it would require to still have an older version of the add-on installed somewhere, but nothing _should_ have changed under the hood that would invalidate the token. Maybe I'll add a check when the popup is opened but I'll have to think about how it could be done without bothering the user nor spaming AL api to ensure the token is valid.

**Comment by Davoleo** [comment]
Fair enough, yeah don't worry about it, I thought it may have been an oversight / a debug leftover, since I don't think it was happening before v2, but it doesn't really bother me.
Normal context menu works just fine, although I've noticed sometimes it just fails to open the tab and you have to repeat the procedure again, nothing too serious though, as the second time it always works.

**Comment by YellowFish085** [comment]
Hello there,

In regards to the error with the invalid token, I've [added a special error notification](https://github.com/YellowFish085/al-search/commit/6fd3b9477d5f80302bc4a651f0e0a0831a6e2acb) in case Anilist returns that the token is invalid instead of the generic error message. Hopefully that'll make it more clear as to what's happening without having to do more queries to Anilist.

I'll try to push a new version on stores during this week or the next!

_Edit_: See release [v2.0.2](https://github.com/YellowFish085/al-search/releases/tag/v2.0.2)
